app/domain/rotation/master_rotator.rb

```diff
@@ -0,0 +1,93 @@
+module Rotation
+  class MasterRotator
+
+    RotatorNotFound = ::Util::ErrorClass.new(
+      "'{0}' is not an installed rotator"
+    )
+
+    # We inject both the rotation_model and the secret_model to ease unit
+    # testing, and also because, even though the rotation query code was thrown
+    # into Secret, it should probably be refactored out into its own model
+    # class.  When that's done, it will require only a one word change here.
+    #
+    def initialize(rotators:,
+                   account:,
+                   rotation_model: Secret,
+                   secret_model: Secret)
+      @rotators = rotators
+      @account = account
+      @rotation_model = rotation_model
+      @secret_model = secret_model
+    end
+
+    def rotate_every(seconds)
+      loop do
+        rotate_all
+        sleep(seconds)
+      end
+    end
+
+    def rotate_all
+      rotations = rotation_model.required_rotations
+      Parallel.each(rotations, in_threads: 10) do |secret|
+
+        rotator = @rotators[secret['rotator']]
+        raise RotatorNotFound unless rotator
```

Comment:
Down the line it might make sense to validate rotators when a policy is imported instead of having a user find a rotator mis-spelling when they try to rotate.

Reply:
👍